### PR TITLE
test(repo): Add integration tests for NextJS quickstart guide

### PR DESCRIPTION
**Pull request by octoper**
## Description

This PR adds integration testing for NextJS quickstart with app router, and also adds some testing utilities.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: integration/models/environment.ts [hunk 1]
```diff
@@ -1,21 +1,43 @@
...
+  get id(): string;
+  setId(newId: string): EnvironmentConfig;
+  setEnvVariable(type: keyof EnvironmentVariables, name: string, value: any): EnvironmentConfig;
+  removeEnvVariable(type: keyof EnvironmentVariables, name: string): EnvironmentConfig;
+  get publicVariables(): EnvironmentVariables['public'];
+  get privateVariables(): EnvironmentVariables['private'];
+  toJson(): { public: Record<string, string>; private: Record<string, string> };
+  fromJson(json: ReturnType<EnvironmentConfig['toJson']>): EnvironmentConfig;
+  clone(): EnvironmentConfig;
+};
```
Comment (panteliselef) [inline integration/models/environment.ts:16]: 💯

**Review by panteliselef — APPROVED** [review]
Great stuff !

@octoper  Quick question, couldn't we have used the next-app-router template and override the files inside the tests instead ? Would that have been more complicated or messy to maintain ?

**Comment by octoper** [comment]
@panteliselef Yeah we could have used it but we decided with @nikosdouvlis to create a fresh one just for the quickstart, but we are probably gonna revisit that.

**Comment by panteliselef** [comment]
I was just curious, no need to revisit just because of my comment.

File: integration/templates/next-app-router-quickstart/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "build": "next build",
+    "dev": "next dev",
+    "lint": "next lint",
+    "start": "next start"
+  },
+  "dependencies": {
+    "@types/node": "^20",
+    "@types/react": "^18",
+    "@types/react-dom": "^18",
+    "next": "14.1.0",
```
[the remark this answers is not in the document]
Comment (octoper) [reply integration/templates/next-app-router-quickstart/package.json:15]: I was just following the other templates, that's why I pinned it but yeah we should not ha the version pinned.

**Review by LekoArts — APPROVED** [review]

**Pull request merged**